### ISSUE TITLE
Extension of `Dendron: Open Link`/`ctrl + click`/`link click in preview` command for other external services

### ISSUE THREAD

**Issue by Mannimu** [dendronhq/dendron#638]
## Please Select if your Request is Either something new or an Enhancement ##

- [x] Enhancement of an existing Feature.
- [ ] Request of a new feature.

## Please select the area your request applys to. (Multiple selections are Possible) ##

- [ ] Onboard. Initial vault setup and import/export pods 
- [ ] Create. Note creation, lookup, snippets and templates
- [x] Retrieve. Backlinks, references, graph view
- [ ] Structure. Refactoring, multi-vault and schemas
- [ ] Publish. Sharing your repo with the world
- [ ] Misc (Choose this if your not sure)

## Is your feature request related to a problem? Please describe ##

In Dendron trying to open an external link like `[Link to external source](integrity://server:port/path/to/asset/inside/service)` results in an error message as shown below when using `Dendron: Open Link` command or `clicking link in preview`. 

`ctrl + click` and VS Code internal command `Open Link` doesn't do anything perceivable.

### Use case:
![image](https://user-images.githubusercontent.com/32539402/115084025-926b6a80-9f08-11eb-95a5-6733fac777ec.png)

### Error when using `Dendron: Open Link`:
![image](https://user-images.githubusercontent.com/32539402/115084058-9ac3a580-9f08-11eb-81cc-bca6ae136685.png)

### Error when `clicking link in preview`:
![image](https://user-images.githubusercontent.com/32539402/115084081-a2834a00-9f08-11eb-8a4f-93241bda39eb.png)

## Describe the solution you'd like ##

I'd like to easily open external links to other services on my system e.g. `[Link to external source](integrity://server:port/path/to/asset/inside/service)` with the common commands `Dendron: Open Link` / `ctrl + click` / `link click in preview` or a `new command to open external links`.
 
Clicking/opening those links in MS OneNote for example works perfectly.

## Describe alternatives you've considered ##

User Experience would be best in my perspective, when all current command `Dendron: Open Link` or `ctrl + click` or `link click in preview` work, but any one of the following commands would suffice: 

* `Dendron: Open Link` 
* `ctrl + click` 
* `link click in preview` 
* `new command to open external links` 

## Additional context ##

...

**Comment by Mannimu** [comment]
Found a workaround for Markdown Preview Enhanced, where other protocolls can be defined in the white list. This way `clicking the link` in the preview works fine. But the other command options `Dendron: Open Link` / `ctrl + click` still don't work.

![MPE_Protocoll_White_List](https://user-images.githubusercontent.com/32539402/115305137-2e45e200-a166-11eb-8fd8-4a66c454e687.png)

**Comment by kpathakota** [comment]
@Mannimu This is a great idea, the way I see it - I think each of the protocols needs a custom handler (I don't know if it's possible to delegate this to the system directly). Are you familiar with how this might work?

**Comment by Mannimu** [comment]
Hi @kpathakota, unfortunately i'm not familiar with the underlying technical aspects. Maybe the above described workaround of MPE could include a hint, how this thing could be solved. 
I suppose MPE does simply parses the links and if those start with the described protocoll prefix, it doesn't preclude the path of the markdown file but leaves the link as it is. Then the clicking of the link in the preview works fine and the underlying system services catch the request.

**Comment by kpathakota** [comment]
Yes, in that case, might be a way of simply removing a restriction in our handling of links. Will look into this.
